bpf-gpl/tc.c

```diff
@@ -607,6 +607,10 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		state.post_nat_ip_dst = nat_dest->addr;
 		state.post_nat_dport = nat_dest->port;
 	} else {
+		if (res == NAT_NO_BACKEND){
+			icmp_v4_port_unreachable(skb);
+			goto allow;
+		}
 		state.post_nat_ip_dst = state.ip_dst;
 		state.post_nat_dport = state.dport;
 	}
```

Comment:
```suggestion
	} else if (res == NAT_NO_BACKEND){
			icmp_v4_port_unreachable(skb);
			goto allow;
	} else {
		state.post_nat_ip_dst = state.ip_dst;
		state.post_nat_dport = state.dport;
	}
```

or perhaps a switch?